lib/events.js

```diff
@@ -119,13 +153,34 @@ function processEvents(req) {
      *
      */
 
+    debugger;
+
+    var x = signer.verify(JSON.stringify(req.body), bs58.decode(headers.signature), bs58.decode(headers.publickey) );
+    var y = signer.verify(JSON.stringify(req.body), signer.util.encodeUTF8(bs58.decode(headers.signature)), signer.util.encodeUTF8(bs58.decode(headers.publickey)) );
+    debugger;
+    return mongo
+        .read(nconf.get('schema').supporters, {
+            userId: _.parseInt(headers.supporterId)
+        })
+        .then(function(supporter) {
+            console.log(_.size(req.rawBody));
+            debugger;
+            debug("%d", _.size(supporter.publicKey));
+
+            if (signer.verify('NaCL is amazing!', signature, publicKey)){
+                    console.log('Signature is valid!');
+            }
+        });
+
+
+        /* TODO: put this code as part of the bluebird chain */
     var processed = _.reduce(_.get(req, 'body'), parseEvents, {
```

Comment:
not not. when the signature validation success, then can be ported in the bluebird pattern